js/src/logger.ts

```diff
@@ -997,18 +997,32 @@ export class Logger<IsAsyncFlush extends boolean> implements Exportable {
   ): PromiseUnless<IsAsyncFlush, R> {
     const { setCurrent, ...argsRest } = args ?? {};
     const span = this.startSpan(argsRest);
-    const ret = runFinally(
-      () => {
-        if (setCurrent ?? true) {
-          return withCurrent(span, callback);
-        } else {
-          return callback(span);
-        }
-      },
-      () => span.end(),
-    );
+
+    let ret;
+    try {
+      ret = runFinally(
+        () => {
+          if (setCurrent ?? true) {
+            return withCurrent(span, callback);
+          } else {
+            return callback(span);
+          }
+        },
+        () => span.end(),
```

Comment:
This seems slightly wrong because we're "end"-ing the span before we catch and log the error. Since `end` just logs an end time this should in theory be benign, but idiomatically we should be running the finally clause after the catch.

Reply:
makes sense